cardano-cli/src/Cardano/CLI/EraBased/Transaction/HashCheck.hs

```diff
@@ -0,0 +1,84 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Cardano.CLI.EraBased.Transaction.HashCheck
+  ( checkCertificateHashes
+  , checkVotingProcedureHashes
+  , checkProposalHashes
+  )
+where
+
+import           Cardano.Api (Certificate (..), ExceptT, firstExceptT)
+import qualified Cardano.Api as L
+import qualified Cardano.Api.Ledger as L
+import qualified Cardano.Api.Shelley as Shelley
+
+import           Cardano.CLI.Run.Hash (carryHashChecks)
+import           Cardano.CLI.Types.Common (MustCheckHash (..), PotentiallyCheckedAnchor (..))
+import           Cardano.CLI.Types.Errors.TxCmdError (TxCmdError (..))
+
+import           Control.Monad (forM_)
+
+-- | Check the hash of the anchor data against the hash in the anchor
+checkAnchorMetadataHash :: L.Anchor L.StandardCrypto -> ExceptT TxCmdError IO ()
+checkAnchorMetadataHash anchor =
+  firstExceptT (TxCmdHashCheckError $ L.anchorUrl anchor) $
+    carryHashChecks
+      ( PotentiallyCheckedAnchor
+          { pcaMustCheck = CheckHash
+          , pcaAnchor = anchor
+          }
+      )
+
+-- | Find references to anchor data and check the hashes are valid
+-- and they match the linked data.
+checkCertificateHashes :: Certificate era -> ExceptT TxCmdError IO ()
+checkCertificateHashes cert = do
+  mAnchor <- L.withExceptT TxCmdPoolMetadataHashError $ L.except $ L.getAnchorDataFromCertificate cert
+  maybe (return mempty) checkAnchorMetadataHash mAnchor
+
+-- | Find references to anchor data in voting procedures and check the hashes are valid
+-- and they match the linked data.
+checkVotingProcedureHashes
+  :: Shelley.ShelleyBasedEra era -> Shelley.VotingProcedures era -> ExceptT TxCmdError IO ()
+checkVotingProcedureHashes eon (Shelley.VotingProcedures (L.VotingProcedures voterMap)) =
+  Shelley.shelleyBasedEraConstraints eon $
+    forM_
+      voterMap
+      ( \vpMap ->
+          forM_
+            vpMap
+            ( \(L.VotingProcedure _ mAnchor) ->
+                forM_ mAnchor checkAnchorMetadataHash
+            )
+      )
```

Comment:
```suggestion
      ( mapM $\(L.VotingProcedure _ mAnchor) ->
                forM_ mAnchor checkAnchorMetadataHash
      )
```
I think mapM may be more readable here

Reply:
Applied in commit: https://github.com/IntersectMBO/cardano-cli/pull/951/commits/d854349759700ad18089ae84f6157e1ba9f0b8b2